sela- started working on general opmode + added servo to turn scoring…

boozmcooz7 opened the pull request:
… unit
אין לי מושג מה עשיתי בטלאופ,חירטטתי פקודות אז שהקוד יפסיק לצרוח עליי
Before issuing a pull request, please see the contributing page.

boozmcooz7 committed:
sela- started working on general opmode + added servo to turn scoring unit

GOBI8450 committed:
Linear Intake teleop back to where he belongs - Teleop folder! my mistake ngl

ori-coval commented on TeamCode/src/main/java/org/firstinspires/ftc/teamcode/CommandGroup/RobotCommands.java:
```diff
@@ -5,30 +5,66 @@
+                new ParallelCommandGroup(
                 mmSystems.intakeArm.intakeDown(),
-                mmSystems.intakEndUnit.openIntakeClaw());
+                mmSystems.intakEndUnit.openIntakeClaw()), new WaitCommand(2*timeLinearIntake));
```
what is the use of `new WaitCommand(2*timeLinearIntake)`?

boozmcooz7 replied:
after the claw is opened i want the command to wait (until sample is absorbed), I don't have a way of ensuring that yet so I put a wait to command

ori-coval commented on TeamCode/src/main/java/org/firstinspires/ftc/teamcode/CommandGroup/RobotCommands.java:
```diff
@@ -5,30 +5,66 @@
                 mmSystems.intakEndUnit.closeIntakeClaw(),
-                mmSystems.intakeArm.intakeUp());
+                mmSystems.intakeArm.intakeUp()).
+                andThen(mmSystems.linearIntake.setPosition(0));
```
replace 0 with a constant

ori-coval commented on TeamCode/src/main/java/org/firstinspires/ftc/teamcode/CommandGroup/RobotCommands.java:
```diff
@@ -5,30 +5,66 @@
+                andThen(mmSystems.linearIntake.setPosition(0));
     }
 
+    /* score sample-
+    1. elevator get to desired height
+    2. scoring servo turn
+    3. scoring claw open
+     */
+    public static Command ScoreSample(){
+        return new SequentialCommandGroup(
+                mmSystems.elevator.moveToPose(Elevator.HIGH_BASKET),
+                mmSystems.scoringEndUnit.scoreScoringServo(),
+                mmSystems.scoringEndUnit.openScoringClaw(),new WaitCommand(timeLinearIntake));
```
why not do it in parallel?

ori-coval commented on TeamCode/src/main/java/org/firstinspires/ftc/teamcode/TeleOp/TeleOp.java:
```diff
@@ -0,0 +1,45 @@
+    @Override
+    public void onInit() {
+        MMRobot.getInstance().mmSystems.initRobotSystems();
+        gamepadEx = new GamepadEx(gamepad1);
```
no need to create a new gamepad take the ones from MMSystems

ori-coval commented on TeamCode/src/main/java/org/firstinspires/ftc/teamcode/TeleOp/TeleOp.java:
```diff
@@ -0,0 +1,45 @@
+        gamepadEx = new GamepadEx(gamepad1);
+
+        //button x-
+        if (gamepadEx.wasJustPressed(GamepadKeys.Button.X)){
+            RobotCommands.ScoreSample();
+            RobotCommands.HoldPosScoring();
+        }
+
+
+        // button y -total intake
+        if (gamepadEx.wasJustPressed(GamepadKeys.Button.Y)){
+            RobotCommands.IntakeCommand();
+            RobotCommands.IntakeDoneCommand();
+        }
```
create triggers

ori-coval commented on TeamCode/src/main/java/org/firstinspires/ftc/teamcode/TeleOp/TeleOp.java:
```diff
@@ -0,0 +1,45 @@
+            RobotCommands.IntakeCommand();
+            RobotCommands.IntakeDoneCommand();
+        }
+        waitForStart();
```
remove this(my bad)

ori-coval commented on TeamCode/src/main/java/org/firstinspires/ftc/teamcode/SubSystems/ScoringEndUnit.java:
```diff
@@ -10,21 +10,30 @@
 
 public class ScoringEndUnit extends SubsystemBase {
     CuttleServo clawscoringServo;
+    CuttleServo posscoringServo;
```
`posscoringServo` did you mean `scoringServo`?

boozmcooz7 replied:
i wanted it to be clear it was the position servo and not the claw

ori-coval replied:
i think scoringArmServo would be a better name

boozmcooz7 committed:
sela- added triggers and fixed issues with General opmode

ori-coval commented on TeamCode/src/main/java/org/firstinspires/ftc/teamcode/CommandGroup/RobotCommands.java:
```diff
@@ -5,30 +5,67 @@
     private static MMSystems mmSystems = MMRobot.getInstance().mmSystems;
+    private static int timeLinearIntake= 500; //millis
+    private static double linearIntakeOpen = 2.22;
+    private  static double linearIntakeClosed = 0;
```
make the constants final

boozmcooz7 committed:
finished writing telop

added button to finalize scoring,intake based off of joystick,formating and convention

ori-coval commented on TeamCode/src/main/java/org/firstinspires/ftc/teamcode/CommandGroup/RobotCommands.java:
```diff
@@ -1,34 +1,71 @@
-    private static MMSystems mmSystems = MMRobot.getInstance().mmSystems;
+    private static final MMSystems mmSystems = MMRobot.getInstance().mmSystems;
+    private static final int timeLinearIntake = 500; //millis
+    private static final double linearIntakeOpen = 2.22;
```
linearIntakeOpen  isnt used remove it

ori-coval commented on TeamCode/src/main/java/org/firstinspires/ftc/teamcode/CommandGroup/RobotCommands.java:
```diff
@@ -1,34 +1,71 @@
+    private static final int timeLinearIntake = 500; //millis
+    private static final double linearIntakeOpen = 2.22;
+    private static final double linearIntakeClosed = 0;
+    private final double maxOpening = 0.23;
```
maxOpening isnt used remove it

ori-coval commented on TeamCode/src/main/java/org/firstinspires/ftc/teamcode/CommandGroup/RobotCommands.java:
```diff
@@ -1,34 +1,71 @@
+    3. intake down and open claw*/
+
+    public static Command IntakeCommand(DoubleSupplier intake_trigger) {
+        return new SequentialCommandGroup(mmSystems.linearIntake.setPositionByJoystick(joysxlj),
```
`joysxlj`?!

also the command need to happen in parallel so the intake end unit would go down and open while the linear intake opens

also currently this wouldnt even work Sequentialy cuse the setPositionByJoystick never ends cuse its a run command

ori-coval commented on TeamCode/src/main/java/org/firstinspires/ftc/teamcode/CommandGroup/RobotCommands.java:
```diff
@@ -1,34 +1,71 @@
+
+    public static Command IntakeCommand(DoubleSupplier intake_trigger) {
+        return new SequentialCommandGroup(mmSystems.linearIntake.setPositionByJoystick(joysxlj),
+                new WaitCommand(timeLinearIntake),
```
because the linearIntake open to a location by the trigger there is no need to wait for it to open so remove this

also like in my erlier comment the this wont work because the setPositionByJoystick neev

ori-coval commented on TeamCode/src/main/java/org/firstinspires/ftc/teamcode/CommandGroup/RobotCommands.java:
```diff
@@ -1,34 +1,71 @@
                 mmSystems.intakEndUnit.closeIntakeClaw(),
-                mmSystems.intakeArm.intakeUp());
+                mmSystems.intakeArm.intakeUp()).
+                andThen(mmSystems.linearIntake.setPosition(linearIntakeClosed));
```
you should wait a bit for the endunit to close before moving it up

also why not have intakeUp and move the linearIntake at the same time it shouldnt be a problem

ori-coval commented on TeamCode/src/main/java/org/firstinspires/ftc/teamcode/SubSystems/ScoringEndUnit.java:
```diff
@@ -10,12 +10,16 @@
     public static double open = 1;
     public static double close = -1;
+    public static double scoring_hold = -1;
+    public static double scoring_score = 1;
```
not clear for what servo this is i would recommand adding arm to the name of this constant

ori-coval replied:
constants should be final

ori-coval commented on TeamCode/src/main/java/org/firstinspires/ftc/teamcode/SubSystems/ScoringEndUnit.java:
```diff
@@ -10,12 +10,16 @@
 
 public class ScoringEndUnit extends SubsystemBase {
     CuttleServo clawscoringServo;
```
`clawscoringServo` upper case S in scoring

ori-coval commented on TeamCode/src/main/java/org/firstinspires/ftc/teamcode/SubSystems/ScoringEndUnit.java:
```diff
@@ -27,4 +31,12 @@ public Command closeScoringClaw() {
+
+    public Command holdScoringServo() {
+        return new InstantCommand(() -> posscoringServo.setPosition(scoring_hold), this);
+    }
```
both functions names are unclear change to something more like

scoringArmScore

ori-coval commented on TeamCode/src/main/java/org/firstinspires/ftc/teamcode/TeleOp/TeleopGeneral.java:
```diff
@@ -0,0 +1,55 @@
+import org.firstinspires.ftc.teamcode.SubSystems.ScoringEndUnit;
+import org.firstinspires.ftc.teamcode.utils.OpModeType;
+
+public class TeleopGeneral extends MMOpMode {
```
`@TeleOp` is missing

ori-coval commented on TeamCode/src/main/java/org/firstinspires/ftc/teamcode/TeleOp/TeleopGeneral.java:
```diff
@@ -0,0 +1,55 @@
+public class TeleopGeneral extends MMOpMode {
+
+
+    public TeleopGeneral(OpModeType.NonCompetition opModeType) {
```
in `public TeleopGeneral(OpModeType.NonCompetition opModeType) {` no need for `OpModeType.NonCompetition opModeType` remove it 

to be clear only remove it in the builder keep the one in the super

ori-coval commented on TeamCode/src/main/java/org/firstinspires/ftc/teamcode/TeleOp/TeleopGeneral.java:
```diff
@@ -0,0 +1,55 @@
+    public void onInit() {
+        MMRobot.getInstance().mmSystems.initRobotSystems();
+
+        Trigger prepareScore = new Trigger(
```
prepare score to where high basket? low basket? specimen?

ori-coval commented on TeamCode/src/main/java/org/firstinspires/ftc/teamcode/TeleOp/TeleopGeneral.java:
on all the triggers add `Trigger` at the end so it would be clear its a trigger

for example: `intakeOpenTrigger`

ori-coval commented on TeamCode/src/main/java/org/firstinspires/ftc/teamcode/TeleOp/TeleopGeneral.java:
```diff
@@ -0,0 +1,55 @@
+                () -> MMRobot.getInstance().mmSystems.gamepadEx1.getButton(GamepadKeys.Button.Y)
+        );
+
+        Trigger totalIntake = new Trigger(
```
why did you add the total? it makes it unclear what the trigger is for

ori-coval commented on TeamCode/src/main/java/org/firstinspires/ftc/teamcode/TeleOp/TeleopGeneral.java:
```diff
@@ -0,0 +1,55 @@
+                () -> MMRobot.getInstance().mmSystems.gamepadEx1.getTrigger(GamepadKeys.Trigger.RIGHT_TRIGGER) > 0.05);
+
+        //button that prepares score- x
+        prepareScore.whileActiveOnce(RobotCommands.PrepareHighSample());
+
+        //new button that score- y
+        scoreSample.whileActiveOnce(MMRobot.getInstance().mmSystems.scoringEndUnit.openScoringClaw());
+
+
+        // intake using trigger
+        totalIntake.whileActiveOnce(
+                RobotCommands.IntakeCommand(() -> MMRobot.getInstance().mmSystems.gamepadEx1.getTrigger(GamepadKeys.Trigger.RIGHT_TRIGGER)).whenFinished(
+                        () -> RobotCommands.IntakeDoneCommand().schedule()));
```
you need to have a meeting of Kanna to choose all the robot function and buttons the driver would have

this need to be done soon so you can know how to create the buttons and commands

boozmcooz7 committed:
sela - remove RR

boozmcooz7 committed:
sela- updated more small changes

boozmcooz7 committed:
sela - chnaged names of variables

GOBI8450 committed:
Added LinearIntake End Unit Rotator. Also added TeleOp for it

boozmcooz7 committed:
sela-connected commands to triggers on controllers

+ refined the commands

GOBI8450 committed:
Merge branch 'master' of https://github.com/MishMash-12016/IntoTheDeep2025Kanna2 into LinearIntakeEndUnitRotator

GOBI8450 committed:
More nice to read rn

boozmcooz7 committed:
sela- updated commands

boozmcooz7 committed:
Merge branch 'master' of https://github.com/MishMash-12016/IntoTheDeep2025Kanna2 into teleopmode

boozmcooz7 committed:
Merge branch 'LinearIntakeEndUnitRotator' of https://github.com/MishMash-12016/IntoTheDeep2025Kanna2 into teleopmode

# Conflicts:
#	TeamCode/src/main/java/org/firstinspires/ftc/teamcode/utils/Configuration.java

boozmcooz7 committed:
added servo to control intake position
